components/src/dynamo/vllm/omni/stage_router.py

```
@@ -101,34 +123,130 @@ async def generate(
+                task.cancel()
+            yield {"error": stage0_output.error, "finished": True}
+            return
+        stage0_text = _stage_output_text(stage0_output)
+
+        final: StageOutput | None = None
+        for stage_idx in range(1, len(self.stage_configs)):
+            try:
+                output = await downstream_tasks[stage_idx]
+            except asyncio.CancelledError:
+                raise
+            except Exception as e:
```

Ruff (0.15.12)

[warning] Do not catch blind exception: `Exception`

(BLE001)